chainstate/test-framework/src/framework_builder.rs

```diff
@@ -35,6 +43,7 @@ pub struct TestFrameworkBuilder {
     chainstate_storage: Store,
     custom_orphan_error_hook: Option<Arc<OrphanErrorHandler>>,
     time_getter: TimeGetter,
+    tx_verification_strategy: TxVerificationStrategy,
```

Comment:
I don't understand the need to make this an enum. Why not just put the object in here then use it in Chainstate, instead of using this enum as a proxy to match on when building?

Reply:
What object exactly? It depends on a strategy and I didn't want to make `TestFrameworkBuilder` generic over it.  And `TransactionVerificationStrategy` trait cannot be used for dynamic polymorphism.

Reply:
Strategy customisation is rare, but if I make it generic then every user of `TestFrameworkBuilder` would have to explicitly specify it. With enums it looks odd, but it is hidden inside the implementation.

Reply:
I see the convenience there, but that match... I bet there's a way to make it go away or abstract it away. I don't know, ignore it for now.